FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -143,6 +143,40 @@ + (void)validateAppOptions:(FIROptions *)appOptions appName:(NSString *)appName
             kFIRLoggerInstallations, kFIRInstallationsMessageCodeInvalidFirebaseAppOptions,
             [missingFields componentsJoinedByString:@", "]];
   }
+
+  [self validateAPIKey:appOptions.APIKey];
+}
+
++ (void)validateAPIKey:(nullable NSString *)APIKey {
+  NSMutableArray<NSString *> *validationIssues = [[NSMutableArray alloc] init];
+
+  if (APIKey.length != kExpectedAPIKeyLength) {
+    [validationIssues addObject:[NSString stringWithFormat:@"API Key length must be %lu characters",
+                                                           (unsigned long)kExpectedAPIKeyLength]];
+  }
+
+  if (![[APIKey substringToIndex:1] isEqualToString:@"A"]) {
+    [validationIssues addObject:@"API Key must start with `A`"];
+  }
+
+  NSCharacterSet *characters = [NSCharacterSet characterSetWithCharactersInString:APIKey];
+  if (![[NSCharacterSet alphanumericCharacterSet] isSupersetOfSet:characters]) {
```

Comment:
Where is `alphanumericCharacterSet` defined?
This is not slower than regex?

Reply:
It is defined in Foundation library (the [docs](https://developer.apple.com/documentation/foundation/nscharacterset/1407466-alphanumerics)) and is pretty fast.